Upgrade to OTel v0.116.0 (#2314)
grafana/alloy: docs/sources/reference/components/otelcol/otelcol.processor.tail_sampling.md

| `num_traces`                  | `int`      | Number of traces kept in memory.                                             | `50000` | no       |
| `expected_new_traces_per_sec` | `int`      | Expected number of new traces (helps in allocating data structures).         | `0`     | no       |
| `decision_cache`              | `object`   | Configures caches for sampling decisions.                                    | `{}`    | no       |

`decision_wait` determines the number of batches to maintain on a channel. Its value must convert to a number of seconds greater than zero.

`num_traces` determines the buffer size of the trace delete channel which is composed of trace ids. Increasing the number will increase the memory usage of the component while decreasing the number will lower the maximum amount of traces kept in memory.

`expected_new_traces_per_sec` determines the initial slice sizing of the current batch. A larger number will use more memory but be more efficient when adding traces to the batch.

`decision_cache` can contain two keys:

> - `sampled_cache_size`: Configures the amount of trace IDs to be kept in an LRU cache,

Reviewer: ```suggestion
- `sampled_cache_size`: Configures the number of trace IDs to be kept in an LRU cache,
```

  persisting the "keep" decisions for traces that may have already been released from memory. 
  By default, the size is 0 and the cache is inactive.

> - `non_sampled_cache_size`: Configures amount of trace IDs to be kept in an LRU cache,

Reviewer: ```suggestion
- `non_sampled_cache_size`: Configures number of trace IDs to be kept in an LRU cache,
```

  persisting the "drop" decisions for traces that may have already been released from memory.
  By default, the size is 0 and the cache is inactive.

You may want to vary the size of the `decision_cache` depending on how many "keep" vs "drop" decisions you expect from your policies. 

> For example, you may allocate a larger `non_sampled_cache_size` if you expect most traces to be dropped.

Reviewer: ```suggestion
For example, you can allocate a larger `non_sampled_cache_size` if you expect most traces to be dropped.
```

> Additionally, when using `decision_cache`, configure it with a much higher value than `num_traces` so decisions for trace IDs are kept longer than the span data for the trace.

Reviewer: ```suggestion
Additionally, when you use `decision_cache`, configure it with a much higher value than `num_traces` so decisions for trace IDs are kept longer than the span data for the trace.
```
